src/schema/v1/filter_artworks.ts

```diff
@@ -384,13 +359,70 @@ const filterArtworksTypeFactory = (
       loader = loaderWithCache
     }
 
-    const blacklistedFields = ["artworks_connection", "__id"]
-    if (queriedForFieldsOtherThanBlacklisted(fieldNodes, blacklistedFields)) {
-      return loader(gravityOptions).then(response =>
-        Object.assign({}, response, { options: gravityOptions })
-      )
+    // Introspect connection args for `artworks_connection`.
+    // Use those and fetch all data in one request.
+    // Resolver for `artworks_connection` just creates a connection
+    // from the already fetched data, instead of re-fetching.
+
+    const connectionArgs = {}
+    info.fieldNodes &&
+      info.fieldNodes.forEach(rootNode => {
+        const visitor = (
+          fieldNode: FieldNode | FragmentDefinitionNode,
+          fragments?: { [key: string]: FragmentDefinitionNode }
+        ): void => {
+          visit(fieldNode, {
+            Field(node, _key, _parent, path, _ancestors) {
+              // Stop recursion for nodes deeper than our threshold
+              if (path.length > SELECTION_DEPTH_THRESHOLD) {
+                return
+              }
+              if (
+                path.length === SELECTION_DEPTH_THRESHOLD &&
+                node.name.value === "artworks_connection"
+              ) {
+                node.arguments &&
+                  node.arguments.forEach(arg => {
+                    if (
+                      ["first", "last", "before", "after"].includes(
+                        arg.name.value
+                      )
+                    ) {
+                      const val = (arg as any).value.value
+                        ? (arg as any).value.value
+                        : 0
```

Comment:
Isn't the whole point of ternary operators to fit on one line? :shakes-fist:

Reply:
(Exceeds Prettier's line length, otherwise it would be on a single line)

Reply:
I know I know but I shake my fist angrily at our bot overlords.

Reply:
Another reason is to capture the result as a `const` instead of a `let`.

Give in to the pretty bots, @joeyAghion, it’s inevitable.

Reply:
@mzikherman  Why is the `any` casting necessary?

Reply:
For some reason, `arg.value`doesn't have a `value`, according to the typing. But logging out the variables shows that it does, and in fact I need it!

`arg` is an `ArgumentNode`, and `arg.value` is a `ValueNode`. I didn't have a chance to look more closely at the typings for those. Generally speaking, I'm using `any` here to get around a 'known' incorrectness in the typing (since there is a `.value` property). It's worth revisiting.